src/Mvc/Mvc.DataAnnotations/test/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -1969,6 +1998,14 @@ public class TypeImplementIInterfaceWithNonNullProperty : AbstraceTypehNonNullPr
     {
         public override string Property { get; set; } = string.Empty;
     }
+
+    public class TypeWithListElements
+    {
+        public List<string> ListWithRequiredStrings { get; set; } = new();
```

Comment:
Probably want to include `string[]`, `IList<string>`, `ICollection<string>`, `IEnumerable<string>` etc.